app.js

```diff
@@ -24,14 +27,30 @@ app.get("/api/articles/:article_id", getArticleByID);
 
 app.get("/api/articles", getArticles);
 
+app.post("/api/articles/:article_id/comments", postComment)
+
 app.all('/*', (req,res) => {
     res.status(404).send({msg: "path not found"})
 })
 
 app.use((err,req,res,next) => {
-    if(err.code === '22P02') {
+    //console.log(err)
+    //console.log(err.code)
+    if(err.status && err.msg) {
+        res.status(err.status).send({msg: err.msg})
+    }
+    if(err.length === 240) {
         res.status(400).send({msg: 'Bad Request'})
     }
+    if(err.constraint === 'comments_article_id_fkey') {
+        res.status(404).send({msg: 'not found'})
+    }
```

Comment:
Line `42` and `45` shouldn't be necessary, I wouldn't advise handling errors by the length... handling by the constraint could be situationally okay 
If you removed both these if statements I believe your tests should all still pass ?